lib/api/controllers/routerController.js

```diff
@@ -418,15 +215,44 @@ module.exports = function RouterController (kuzzle) {
     api.get('/', (request, response) => {
       var _routes = {};
 
-      extractRouteInfo(_routes, routes);
-      extractRouteInfo(_routes, kuzzle.pluginsManager.routes, '/_plugin');
+      extractRouteInfo(_routes, routes, '/' + kuzzle.config.apiVersion);
+      extractRouteInfo(_routes, [{verb: 'get', url: '/_serverInfo', controller: 'read', action: 'serverInfo'}]);
+      extractRouteInfo(_routes, kuzzle.pluginsManager.routes, '/' + kuzzle.config.apiVersion + '/_plugin');
 
       response.setHeader('Access-Control-Allow-Origin', '*');
       response.setHeader('Access-Control-Allow-Headers', 'X-Requested-With');
       response.writeHead(200, {'Content-Type': 'application/json'});
       response.end(stringify({status: 200, error: null, result: {message: 'Available routes for this API version by verb.', routes: _routes}}));
     });
 
+    apiBase.get('/swagger.json', (request, response) => {
+      var content;
+      try {
+        content = fs.readFileSync(__dirname + '/../../../kuzzle-swagger.json');
+      }
+      catch (e) {
+        content = stringify({error: 'The swagger file has not been generated. Use the `kuzzle swagger` CLI command to generate it.', status: 500});
```

Comment:
well, the goal here, was to respond to those: 
- should the swagger file include routes declared by plugins ? (we ( @benoitvidis and I) answered yes)
- should the swagger file be generated on demand (via http get) ? (we answered no)
- should the swagger files be distributed with Kuzzle ? (we answered no again)

The idea behind the CLI way of generating the swagger files was to allow anyone to generate a  file reflecting its very own Kuzzle environment (with plugins path) and to generate it when it is useful.

Since not anyone will use those swagger files at the moment, and since it is really useful to generate them when an administrator (de)activates plugins, this way looked a good way to do this.

Another option may be to add a specific route for admins only in order to generate the swagger files (from the BO for instance).